test(minifier): merge variable declarations into for statement initializers

sapphi-red opened the pull request:
Added some test cases that describes that the transformations are safe, which I misunderstood once.

Boshen commented:
### Merge activity

* **Sep 15, 5:59 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
test(minifier): merge variable declarations into for statement initializers (#13770)

Added some test cases that describes that the transformations are safe, which I misunderstood once.
